sim/SparkBIDS.py

```diff
@@ -36,48 +36,54 @@ def run(self, sc):
             # rdd[0] is the participant label, rdd[1] is the data (if HDFS) or None
 
             # Participant analysis (done for all apps)
-            mapped = rdd.filter(lambda x: self.get_participant_from_fn(x[0]) not in self.skipped_participants)\
-                        .map(lambda x: self.run_participant_analysis(self.get_participant_from_fn(x[0]),
-                                                                     x[1]))
-
+            mapped = rdd.filter(lambda x: self.get_participant_from_fn(x[0]) not in self.skipped_participants) \
+                .map(lambda x: self.run_participant_analysis(self.get_participant_from_fn(x[0]),
+                                                             x[1]))
+            self.participant_analysis_failed=False
             for result in mapped.collect():
                 self.pretty_print(result)
+                if self.check_failure(result):
+                    #Disable Group Analysis if Participant Analysis Fails
+                    self.do_group_analysis=False
+                    self.participant_analysis_failed=True
 
+        if self.participant_analysis_failed:
+            print("Error# Participant analysis failed. Aborting group analysis")
```

Comment:
you can print this message in the previous if block (line 48). Then you don't need `self.participant_analysis_failed`. Print the message only if `self.do_group_analysis`.